app-android/src/main/java/com/ustadmobile/port/android/view/ContentEntryEdit2Fragment.kt

```diff
@@ -221,6 +229,29 @@ class ContentEntryEdit2Fragment(private val registry: ActivityResultRegistry? =
             entity?.language = language
             entity?.primaryLanguageUid = language.langUid
         }
+
+        navController.currentBackStackEntry?.savedStateHandle?.observeResult(this,
+                ImportedContentEntryMetaData::class.java) {
+            val metadata = it.firstOrNull() ?: return@observeResult
+            // back from navigate import
+            entryMetaData = metadata
+            when (entryMetaData) {
+                null -> {
```

Comment:
Metadata cannot be null (because it would have returned already). This when statement is therefor redundant. The user should have already received an error on the previous screen if we did not support the link.